base/task.jl

```diff
@@ -818,6 +818,7 @@ end
 end
 
 function wait()
+    GC.safepoint()
```

Comment:
not sure if this is the right place to put this, but it prevents the shell from hanging when we pasted in the simple program in the linked issue

Reply:
There is an unconditional safepoint in process_events() after the yield, but I suppose if the code is not returning to an existing Task, we would not reach that?

Reply:
Oh, interesting, good to know! I see it now, here:

https://github.com/JuliaLang/julia/blob/5f182852e86c02c6371c913741376cc21c886e25/src/jl_uv.c#L210

Was that added after Julia 1.6? We were experiencing some long pauses in prod on 1.6, which is what got us looking into this. But seems like it's already been addressed on master?

In any case, as you say, it seems like pausing for GC _before_ task yielding would give slightly more opportunities for GC to run, but i agree that it doesn't seem much different.